src/utils/content_protection_utils/src/content_protection_utils.c

```diff
@@ -0,0 +1,51 @@
+/**
+ * @file content_protection_utils.c
+ * @brief Provides an implementation of the utilities for interacting with the content protection blocks
+ *
+ * @copyright Copyright (c) Microsoft Corporation.
+ * Licensed under the MIT License.
+ */
+#include "content_protection_utils.h"
+#include "content_protection_field_defs.h"
+#include <azure_c_shared_utility/strings.h>
+#include <parson.h>
+#include <stdbool.h>
+#include <strings.h>
+
+bool ContentProtectionUtils_DecryptDek(const STRING_HANDLE dek, const STRING_HANDLE dekCryptAlg);
+
+DecryptionAlg ContentProtectionUtils_GetDecryptAlgFromDecryptionInfo(const JSON_Object* decryptInfo)
```

Comment:
I was trying to obscure the parsing of the JSON into the content_protection_utils so we aren't adding new fields/defines to the adu_core_interface but maybe that should be a part of the adu_core_interface anyway since these fields are apart of that interface.

I'll move the field definitions to adu_core_interface and change this to the suggested inputs.

For the KeyLength the current schema has it as a string. I'm gunna keep that as a char*

Reply:
From looking it over there doesn't seem to be an easy way place to add these field definitions. Do we have a recommended place for these? If not I'll keep these definitions where they are.